tools/SetupFlow/DevHome.SetupFlow/ViewModels/ConfigurationFileViewModel.cs

```diff
@@ -97,46 +100,71 @@ public async Task ConfigureAsNonAdminAsync()
     /// <returns>True if a YAML configuration file was selected, false otherwise</returns>
     public async Task<bool> PickConfigurationFileAsync()
     {
-        // Get the application root window.
-        var mainWindow = Application.Current.GetService<WindowEx>();
-
         // Create and configure file picker
         Log.Logger?.ReportInfo(Log.Component.Configuration, "Launching file picker to select configuration file");
-        var file = await mainWindow.OpenFilePickerAsync(Log.Logger, ("*.yaml;*.yml", StringResource.GetLocalized(StringResourceKey.FilePickerFileTypeOption, "YAML")));
+        var file = await _mainWindow.OpenFilePickerAsync(Log.Logger, ("*.yaml;*.yml;*.winget", StringResource.GetLocalized(StringResourceKey.FilePickerFileTypeOption, "YAML")));
+        return await LoadConfigurationFileInternalAsync(file);
+    }
 
+    /// <summary>
+    /// Load a configuration file if feature is enabled
+    /// </summary>
+    /// <param name="file">The configuration file to load</param>
+    /// <returns>True if the configuration file was loaded, false otherwise</returns>
+    public async Task<bool> LoadFileAsync(StorageFile file)
+    {
+        Log.Logger?.ReportInfo(Log.Component.Configuration, "Loading a configuration file");
+        if (!await _dsc.IsUnstubbedAsync())
+        {
+            await _mainWindow.ShowErrorMessageDialogAsync(
+                StringResource.GetLocalized(StringResourceKey.ConfigurationViewTitle, file.Name),
+                "Configuration is disabled. Attempting to enable it. Please try again in a few minutes",

```

Comment:
Adding @joadoumie for text suggestion

Reply:
"Winget configuration is currently disabled on your machine. We are initiating the enablement process, which may take a few minutes. Please try again soon."